qt/python/mantidqt/widgets/fitpropertybrowser/addfunctiondialog/test/test_addfunctiondialogpresenter.py

```diff
@@ -0,0 +1,45 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#     NScD Oak Ridge National Laboratory, European Spallation Source
+#     & Institut Laue - Langevin
+# SPDX - License - Identifier: GPL - 3.0 +
+#  This file is part of the mantid workbench.
+
+import unittest
+
+from mantid.py3compat.mock import MagicMock, patch
+from mantidqt.utils.qt.testing import start_qapplication
+from mantidqt.widgets.fitpropertybrowser.addfunctiondialog.presenter import AddFunctionDialog
+from testhelpers import assertRaisesNothing
+
+
+@start_qapplication
```

Comment:
Could we consider using patching `mantidqt.widgets.fitpropertybrowser.addfunctiondialog.presenter.AddFunctionDialogView` instead of using `start_qapplication`? Now the MVP has been implemented we should be able to patch out the Qt view completely and then starting `QApplication` is not necessary - this will make the test more reliable too as some of the random test failures on the server are due to this `start_qapplication` thing. My apologies if I confused things by mentioning it the first comment but it was meant to be an either/or option but I explained myself poorly.